stage1/reaper/reaper.sh

```diff
@@ -2,17 +2,31 @@
 shopt -s nullglob
 
 SYSCTL=/usr/bin/systemctl
+DIAG=/diagnostic
 
 if [ $# -eq 1 ]; then
     app=$1
     status=$(${SYSCTL} show --property ExecMainStatus "${app}.service")
-    echo "${status#*=}" > "/rkt/status/$app"
-    if [ "${status#*=}" != 0 ] ; then
+    status="${status#*=}"
+    echo "${status}" > "/rkt/status/$app"
+    if [ "${status}" != 0 ] ; then
         # The command "systemctl exit $status" sets the return value that will
         # be used when the pod exits (via shutdown.service).
         # This command is available since systemd v227. On older versions, the
         # command will fail and rkt will just exit with return code 0.
         ${SYSCTL} exit ${status#*=} 2>/dev/null
+        # systemd EXIT_EXEC is 203 (https://github.com/systemd/systemd/blob/v230/src/basic/exit-status.h#L47)
+        if [ "${status}" == 203 ]; then
+            # Run diagnostic
+            unit_file=$(${SYSCTL} cat "${app}.service")
+            target=${unit_file##*ExecStart=}
```

Comment:
Because the output of `systemctl show --property ExecStart` is more complicated to parse.